### PR TITLE
rust: kernel: clean `rustdoc` warnings

### PR DESCRIPTION
**Pull request by ojeda**
e.g. https://github.com/Rust-for-Linux/linux/pull/402/checks#step:60:27

### CODE REVIEW

**Comment by ksquirrel** [comment]
Review of `cfdc1c98d50c`:
  - ✔️ Commit [`cfdc1c9`](https://github.com/Rust-for-Linux/linux/commit/cfdc1c98d50cb67aa4a9ca91a6925160a8420a25): Looks fine!

**Comment by wedsonaf** [comment]
#410 introduces another one of these.

**Comment by ksquirrel** [comment]
Review of `4b832b45b818`:
  - ✔️ Commit [`4b832b4`](https://github.com/Rust-for-Linux/linux/commit/4b832b45b8188697edf5bb7c57128293ef5b4538): Looks fine!

**Comment by ojeda** [comment]
Thanks for the heads up -- merged #410 and fixed here.

**Review by wedsonaf — APPROVED** [review]

File: rust/kernel/file.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@ use core::{mem::ManuallyDrop, ops::Deref};
 ///
 /// # Invariants
 ///
-/// The pointer [`File::ptr`] is non-null and valid. Its reference count is also non-zero.
+/// The pointer `File::ptr` is non-null and valid. Its reference count is also non-zero.
```
Comment (bjorn3) [inline rust/kernel/file.rs:15]: I think you can use `File.ptr` as link.

Comment (ojeda) [reply rust/kernel/file.rs:15]: That works, but the link appears still as "[`File.ptr`]" i.e. it is still broken because it points to a private field. :-(

Comment (ojeda) [reply rust/kernel/file.rs:15]: The other day we were discussing whether the invariants should be public or hide them, or whether we wanted to have private fields shown. Not sure what we will do.

Comment (bjorn3) [reply rust/kernel/file.rs:15]: Of course. It is private. Maybe say "The inner pointer" as you can't see the `ptr` field in the docs?

Comment (ojeda) [reply rust/kernel/file.rs:15]: Yeah, that is another option, definitely.

Comment (wedsonaf) [reply rust/kernel/file.rs:15]: > Of course. It is private. Maybe say "The inner pointer" as you can't see the `ptr` field in the docs?

That is sort of I what I did for rbtrees: https://github.com/Rust-for-Linux/linux/pull/400/files#diff-0cb585b4b0d3d30535d603fd0ae07753fcd5a180b13d8f9f06164093c4551722R40

**Pull request merged**